src/companies/controllers/getCompanyUsers.ts

```diff
@@ -1,25 +1,98 @@
-import type { Request, Response, NextFunction } from "express";
+import type { Request, Response, NextFunction } from 'express';
+import prisma from '../../config/prisma.js';
 
-async function getCompanyUsers(req: Request, res: Response, next: NextFunction) {
+// 지원하는 검색 키
+const SEARCHABLE_FIELDS = [
+  'name',
+  'email',
+  'employeeNumber',
+  'phoneNumber',
+] as const;
+type SearchBy = (typeof SEARCHABLE_FIELDS)[number];
```

Comment:
이미 const로 선언된 상수를 한번 더 as로 const로 하는건 무의미한 코드에요, typeof도 아마 필요 없을거에요

위랑 별개로 이런걸 Enum(열거형) 이라고 하는데요. 자바스크립트나 타입스크립트에서 열거형을 사용할 땐 이렇게 사용해요

```suggestion
const SEARCHABLE_FIELDS = {
  NAME: "name",
  EMAIL: "email",
  EMPLOYEENUMBER: "employeeNumber",
  PHONENUMBER: "phoneNumber",
};

SEARCHABLE_FIELDS.NAME;
```

이렇게 대문자 키로 이루어진 객체로 만들어서 꺼내쓰시면 됩니다.